### PR TITLE
New ad-free trial switch and cookie and modified evaluation at sign-in.

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
We created a new master switch for ad-free. This isn't necessary but brings the terminology in to line with current direction (membership -> subscription)

We use a new cookie, to make sure there's no contamination from previous trial participants, and the contents of the cookie have changed to allow it to expire in the same way as other valid user credential identifiers.

The code to inspect the cookie has been strengthened to cope with unexpected data content.

The old cookie is removed at the earliest possible point (`bootstraps/standard/main.js`). This also checks the master switch state and removes the _new_ cookie too, if it's turned off.

We only persist the ad-free status in the cookie if the master switch is **on** and the user is a trial participant.

Updated tests check that a stale cookie causes a data refresh, and that non-participating users don't have the cookie at all.

## What is the value of this and can you measure success?
Regaining control of reduced ads trial.

## Does this affect other platforms - Amp, Apps, etc?
There's a companion PR in Fastly (https://github.com/guardian/fastly-edge-cache/pull/783).
Other than that, it's web only.

## Screenshots
N/A

## Tested in CODE?
Yes

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/user-features.spec.js [hunk 1]
```diff
@@ -143,7 +156,7 @@ describe('The isPayingMember getter', () => {
     it('Is false when the user is logged out', () => {
         jest.resetAllMocks();
         isUserLoggedIn.mockReturnValue(false);
-        expect(isAdFreeUser()).toBe(false);
+        expect(isPayingMember()).toBe(false);
```
Comment (JustinPinner) [inline static/src/javascripts/projects/commercial/modules/user-features.spec.js]: @gidsg I think you were the last to work on this test. I assume this is what you'd intended?

File: static/src/javascripts/projects/commercial/modules/user-features.spec.js [hunk 2]
```diff
@@ -174,7 +187,7 @@ describe('The isInBrexitCohort getter', () => {
     it('Is false if the user is logged out', () => {
         jest.resetAllMocks();
         isUserLoggedIn.mockReturnValue(false);
-        expect(isAdFreeUser()).toBe(false);
+        expect(isInBrexitCohort()).toBe(false);
```
Comment (JustinPinner) [inline static/src/javascripts/projects/commercial/modules/user-features.spec.js]: @lmath I think you were the last to work on this test. I assume this is what you'd intended?

Comment (lmath) [inline static/src/javascripts/projects/commercial/modules/user-features.spec.js]: Yes! Thanks for fixing that.

**Review by davidfurey — APPROVED** [review]
Just a few minor points that aren't blockers

File: static/src/javascripts/bootstraps/standard/main.js [hunk 3]
```diff
@@ -177,6 +177,14 @@ const bootStandard = (): void => {
     // Set adtest query if url param declares it
     setAdTestCookie();
 
+    // Remove the old ad-free cookie - we can delete this line in a few days
+    removeCookie('GU_AFU');
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/standard/main.js]: what happens if return after this has been removed and still have the old cookie?

Comment (JustinPinner) [inline static/src/javascripts/bootstraps/standard/main.js]: We're not responding to that cookie anywhere else now. I merged a Fastly change a short while ago to ignore it. This is more about tidying up really, as the new cookie will take precedence. I was wondering if this is the best place for this kind of thing to be done - early enough in the process, and presumably widely executed?

Comment (sndrs) [inline static/src/javascripts/bootstraps/standard/main.js]: yeah seems a good place to me 👍

**Comment by JustinPinner** [comment]
It went to PROD bundled up with another merge. But looks ok so far. The clue is in the renamed master switch ;)

**Pull request merged**